### DISCUSSION TITLE
stopmoving()사용시 월드가 멈추는 현상

### DISCUSSION THREAD

**Discussion by jiwoo50** [naverz/zepeto-studio-kor#1114] [5. Q&A]
플레이어가 오브젝트에 닿으면 일정시간 동안 stopmoving()을 사용해 움직이지 않도록 하였습니다.
하지만 이 stopmoving을 사용하는 동안에 플레이어 뿐만 아니라 월드가 멈추는 현상을 발견했습니다.
어떻게 하면 월드가 멈추지 않고 플레이어만 멈추게 할수있을까요?

아래는 멈추는 코드입니다.
![image](https://user-images.githubusercontent.com/72139485/182508103-75cb1884-aa70-41bc-8ce8-ce7ba4e22923.png)

**Comment by hawai0622** [comment]
timer의 값이 5000이 되기 전까지 while문이 실행되기 때문에 멈추게 됩니다.

**Reply by Deytooh** [reply]
Coroutine 쓰면 문제 해결 할 수 있어요 :) while 쓰면 코드 다 멈추는거예요

**Comment by smlee2952** [comment]
private * StopMoving() {
    yield new UnityEngine.WaitForSeconds(5);
    // Zepeto Character StopMoving();
}

위 함수를 this.StartCoroutine(this.StopMoving()); 이런식으로 호출해주시면 됩니다.
while에서 기다리게 하시면 윗분들 말처럼 전부 멈추게 됩니다.
